yaptide/utils/sim_utils.py

```diff
@@ -33,19 +36,90 @@ def pymchelper_output_to_json(estimators_dict: dict, dir_path: Path) -> dict:
+    """
+    Convert payload data to dictionary with filenames and contents for Editor type projects
+    Otherwise return empty dictionary
+    """
+    conv_parser = get_parser_from_str(parser_type)
+    files_dict = run_parser(parser=conv_parser, input_data=editor_dict)
+    return files_dict
+
+
+def check_and_convert_payload_to_files_dict(payload_dict: dict) -> dict:
     """
-    Function used to write input files to output directory.
-    Returns dictionary with filenames as keys and their content as values
+    Convert payload data to dictionary with filenames and contents for Editor type projects
+    Otherwise return empty dictionary
     """
-    if "beam.dat" not in json_data["sim_data"]:
-        conv_parser = get_parser_from_str(json_data["sim_type"])
-        return run_parser(parser=conv_parser, input_data=json_data["sim_data"], output_dir=output_dir)
-
-    for key, file in json_data["sim_data"].items():
-        with open(Path(output_dir, key), "w") as writer:
-            writer.write(file)
-    return json_data["sim_data"]
+    files_dict = {}
+    json_type = get_json_type(payload_dict)
+    if json_type == JSON_TYPE.Editor:
+        files_dict = convert_editor_payload_to_dict(editor_dict=payload_dict["sim_data"],
+                                                    parser_type=payload_dict["sim_type"])
+    else:
+        logging.warning("Project of %s used, conversion works only for Editor projects", json_type)
+    return files_dict
+
+
+def editor_dict_with_adjusted_primaries(payload_editor_dict: dict) -> dict:
+    """Replace number of primaries in editor dict"""
+    editor_dict = copy.deepcopy(payload_editor_dict['sim_data'])
+    editor_dict['beam']['numberOfParticles'] //= payload_editor_dict['ntasks']
+    return editor_dict
+
+
+def files_dict_with_adjusted_primaries(payload_files_dict: dict) -> dict:
+    """Replace number of primaries in files dict"""
+    files_dict = copy.deepcopy(payload_files_dict['sim_data'])
+    all_beam_lines: list[str] = files_dict['beam.dat'].split('\n')
+    all_beam_strings_with_nstat = [line for line in all_beam_lines if 'NSTAT' in line]
+    if len(all_beam_strings_with_nstat) != 1:
+        return files_dict
+    old_nstat: str = all_beam_strings_with_nstat[0].split()[1]
+    new_nstat = str(int(old_nstat) // payload_files_dict['ntasks'])
+    for i in range(len(all_beam_lines)):
+        if 'NSTAT' in all_beam_lines[i]:
+            all_beam_lines[i] = all_beam_lines[i].replace(old_nstat, new_nstat, 1)
+    files_dict['beam.dat'] = '\n'.join(all_beam_lines)
+    # number_of_tasks = payload_files_dict['ntasks']  -> to be implemented in UI
+    # here we manipulate the files_dict['beam.dat'] file to adjust number of primaries
+    # we manipulate content of the file, no need to write the file to disk
+    print(files_dict['beam.dat'])
+    return files_dict
+
+
+def dict_with_adjusted_primaries(payload_dict: dict) -> dict:
```

Comment:
Should this be `files_dict_with_adjusted_primaries` ?

Reply:
It is correct

Reply:
There are 3 methods:
- `dict_with_adjusted_primaries` - returns `files_dict` for `payload_dict`
- `editor_dict_with_adjusted_primaries` - returns `files_dict` for `payload_editor_dict`
- `files_dict_with_adjusted_primaries` - returns `files_dict` for `payload_files_dict`

Reply:
If it returns `file_dict` then `file_dict` should be in method name. otherwise it's not clear from method name wtmhat it does

Reply:
Method name should encode input and output type